google/cloud/storage/client.py

```diff
@@ -932,12 +932,22 @@ def create_bucket(
 
         """
         bucket = self._bucket_arg_to_bucket(bucket_or_name)
+        query_params = {}
 
         if project is None:
             project = self.project
 
-        if project is None:
-            raise ValueError("Client project not set:  pass an explicit project.")
+        # Use no project if STORAGE_EMULATOR_HOST is set
+        if _get_storage_host().find("storage.googleapis.com") < 0:
+            if project is None:
+                project = _get_environ_project()
+            if project is None:
+                project = "<none>"
+
+        # Only include the project parameter if a project is set.
+        # If a project is not set, falls back to API validation (BadRequest). Removes client-side validation.
```

Comment:
"Removes client-side validation" may be a confusing comment here when read in the future, without the context of this PR.

Reply:
Makes sense, revised.